### PR TITLE
Anyscale integration with LlamaIndex through LangChain 

### PR DESCRIPTION
**Pull request by kylehh**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Disiok** [comment]
Hey @kylehh, thanks for the contribution here! 

Would you like to add this as an native LLM integration instead of using the langchain adapter?
This helps make it more maintainable and less prone to breakage from langchain side.

**Comment by kylehh** [comment]
Thanks for checking @Disiok 
For now [Anyscale Endpoint](https://app.endpoints.anyscale.com/) is not providing its own Python API but leverage OpenAI or LangChain Python API for easy usage and migration
That's why I did't provide a native LLM integration for now.

**Comment by logan-markewich** [comment]
@kylehh I think what @Disiok is getting at is we would prefer a native implementation rather than bringing in langchains LLM.

It's a tiny bit more work, but it really makes development/maintaining it easier for us :)

You can see other LLM integrations like [PaLM](https://github.com/jerryjliu/llama_index/blob/main/llama_index/llms/palm.py), [Replicate](https://github.com/jerryjliu/llama_index/blob/main/llama_index/llms/replicate.py), or [Xinference](https://github.com/jerryjliu/llama_index/blob/main/llama_index/llms/xinference.py) as examples

**Comment by kylehh** [comment]
Understood. and the native implementation added

File: llama_index/llms/anyscale.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    from pydantic.v1 import Field, PrivateAttr
+except ImportError:
+    from pydantic import Field, PrivateAttr
+
+import importlib
+
+from llama_index.callbacks import CallbackManager
+from llama_index.llms import anyscale_utils
+
+importlib.reload(anyscale_utils)
```
Comment (logan-markewich) [inline llama_index/llms/anyscale.py:15]: is this needed?

**Review by logan-markewich — COMMENTED** [review]
Looks good so far! I suspect CI will complain about some CI issues

We use black, ruff, and mypy for linting

You can fix most issues by doing

```
pip install -r requirements.txt
black .
ruff . --fix
```

Also, can you add a sample notebook to `docs/examples/llms` ?

File: llama_index/llms/anyscale.py (hunk 1 above)
Comment (kylehh) [reply llama_index/llms/anyscale.py]: removed

**Comment by logan-markewich** [comment]
Thanks @kylehh !

**Pull request merged**